packages/apollo-server-core/src/ApolloServer.ts

```diff
@@ -942,12 +944,15 @@ export class ApolloServerBase<
    * `{req: express.Request, res: express.Response }` object) and to keep it
    * updated as you upgrade Apollo Server.
    */
-  public async executeOperation(
+  public async executeOperation<
+    TData = Record<string, any>,
+    TVariables = VariableValues,
+  >(
     request: Omit<GraphQLRequest, 'query'> & {
       query?: string | DocumentNode;
-    },
+    } & { variables?: TVariables | VariableValues },
     integrationContextArgument?: ContextFunctionParams,
-  ) {
+  ): Promise<GraphQLResponse & { data?: TData | null }> {
```

Comment:
Do we want to `Omit` data from GraphQLResponse like we do with query above? Or was that actually a pointless thing to do for query too?